cli/src/parse.rs

```diff
@@ -393,6 +526,177 @@ pub fn parse_file_at_path(parser: &mut Parser, opts: &ParseFileOptions) -> Resul
     })
 }
 
+const fn escape_invisible(c: char) -> Option<&'static str> {
+    Some(match c {
+        '\n' => "\\n",
+        '\r' => "\\r",
+        '\t' => "\\t",
+        '\0' => "\\0",
+        '\\' => "\\\\",
+        '\x0b' => "\\v",
+        '\x0c' => "\\f",
+        _ => return None,
+    })
+}
+
+fn render_node_text(source: &str) -> String {
+    let mut escaped = String::new();
+    escaped.reserve(source.len());
+
+    for c in source.chars() {
+        if let Some(esc) = escape_invisible(c) {
+            escaped.push_str(esc);
+        } else {
+            escaped.push(c);
+        }
+    }
+
+    escaped
+}
+
+#[allow(clippy::too_many_arguments)]
+fn write_node_text(
+    opts: &ParseFileOptions,
+    stdout: &mut StdoutLock<'static>,
+    cursor: &TreeCursor,
+    source: &str,
+    color: Option<impl Into<Color> + Copy>,
+    quote: char,
+    row_width: usize,
+    indent_level: usize,
+) -> Result<()> {
+    if source.find('\n') == Some(source.len() - 1)
+        || source.find('\n').is_none()
+        || !cursor.node().is_named()
+    {
+        write!(
+            stdout,
+            "{quote}{}{quote}",
+            paint(color, &render_node_text(source))
+        )?;
+    } else {
+        for line in source.split_inclusive('\n') {
+            if line.is_empty() {
+                break;
+            }
+            if !opts.no_ranges {
+                write!(
+                    stdout,
+                    "\n{}{}{quote}{}{quote}",
+                    render_node_range(opts, cursor, row_width),
+                    "  ".repeat(indent_level + 1),
+                    &paint(color, &render_node_text(line)),
+                )?;
+            } else {
+                write!(
+                    stdout,
+                    "\n{}{quote}{}{quote}",
+                    "  ".repeat(indent_level + 1),
+                    &paint(color, &render_node_text(line)),
+                )?;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn render_node_range(opts: &ParseFileOptions, cursor: &TreeCursor, row_width: usize) -> String {
+    let node = cursor.node();
+    let is_named = node.is_named();
```

Comment:
you called `node.is_named()` in the caller of this function - just pass the value of the first call into here instead.